test/internal/pbxproj_partials/write_bazel_dependencies_tests.bzl

```diff
@@ -0,0 +1,311 @@
+"""Tests for `pbxproj_partials.write_bazel_dependencies`."""
+
+load("@bazel_skylib//lib:unittest.bzl", "asserts", "unittest")
+
+# buildifier: disable=bzl-visibility
+load("//xcodeproj/internal:pbxproj_partials.bzl", "pbxproj_partials")
+
+_OUTPUT_DECLARED_FILE = "a_generator_name_pbxproj_partials/bazel_dependencies"
+_POST_BUILD_DECLARED_FILE = "a_generator_name_pbxproj_partials/post_build_script"
+_PRE_BUILD_DECLARED_FILE = "a_generator_name_pbxproj_partials/pre_build_script"
+
+def _write_bazel_dependencies_test_impl(ctx):
+    env = unittest.begin(ctx)
+
+    # Arrange
+
+    args = []
+
+    def _args_add_all(flag, values, *, map_each = None):
+        args.append(flag)
+        if map_each:
+            args.extend(["{}({})".format(map_each, value) for value in values])
+        else:
+            args.extend(values)
+
+    use_param_file_args = {}
+
+    def _args_use_param_file(param_file):
+        use_param_file_args["use_param_file"] = param_file
+
+    set_param_file_format_args = {}
+
+    def _args_set_param_file_format(format):
+        set_param_file_format_args["format"] = format
+
+    action_args = struct(
+        add = lambda *x: args.extend(x),
+        add_all = _args_add_all,
+        use_param_file = _args_use_param_file,
+        set_param_file_format = _args_set_param_file_format,
+    )
+
+    run_args = {}
+
+    def _action_run(*, arguments, inputs, outputs, **_kwargs):
+        run_args["arguments"] = arguments
+        run_args["inputs"] = inputs
+        run_args["outputs"] = outputs
+
+    declared_files = {}
+
+    def _actions_declare_file(path):
+        declared_files[path] = None
+        return path
+
+    writes = {}
+
+    def _actions_write(write_output, args):
+        writes[write_output] = args
+
+    actions = struct(
+        args = lambda: action_args,
+        declare_file = _actions_declare_file,
+        run = _action_run,
+        write = _actions_write,
+    )
```

Comment:
After one more set of tests that duplicate this, I'll feel good about pulling it out into a test module that can be reused.